Python/Default.sublime-keymap

```diff
@@ -22,4 +22,13 @@
             { "key": "eol_selector", "operator": "not_equal", "operand": "string.quoted.single - punctuation.definition.string.end", "match_all": true }
         ]
     },
+
+    // Helper keybinding for interpolation.
+    { "keys": ["{"], "command": "insert_snippet", "args": {"contents": "{$0}"}, "context":
+        [
+            { "key": "setting.auto_match_enabled" },
+            { "key": "selection_empty", "match_all": true },
+            { "key": "selector", "operator": "equal", "operand": "source.python, meta.string", "match_all": true },
```

Comment:
By using `,` you literly enable this key binding for all strings in all syntaxes.

On the other hand normal key bindings already work once `string` is cleared within the interpolation.

```py
s = f"before { a + b } after"
#            ^^^^^^^^^ - string -> auto-pair already works within that block and should do so like in the rest of the code.
```

Hence I'd suggest a selector such as `source.python string`.

We may also want to not trigger auto-pairing in case of typing escaped braces such as `\{` or `\\\{` etc.

```
{ "key": "preceding_text", "operator": "regex_contains", "operand": "((?<![\\\\])([\\\\]{2})*)$", "match_all": true },
```

Reply:
Thanks. Not sure why I put that comma anyways :/
Good catch on the escaped braces.